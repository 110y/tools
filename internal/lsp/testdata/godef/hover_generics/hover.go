package hover

type value[T any] struct { //@mark(value, "value"),hoverdef("value", value)
	val T   //@mark(valueTparam, "T"),hoverdef("T", valueTparam)
	Q   int //@mark(valueQfield, "Q"),hoverdef("Q", valueQfield)
}

type Value[T any] struct {
	val T   //@mark(ValueTparam, "T"),hoverdef("T", ValueTparam)
	Q   int //@mark(ValueQfield, "Q"),hoverdef("Q", ValueQfield)
}
