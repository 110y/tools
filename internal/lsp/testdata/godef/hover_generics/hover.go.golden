-- value-hoverdef --
```go
type value[T any] struct {
	val T   //@mark(valueTparam, "T"),hoverdef("T", valueTparam)
	Q   int //@mark(valueQfield, "Q"),hoverdef("Q", valueQfield)
}
```
-- valueTparam-hoverdef --
```go
type value[T any] struct {
	val T   //@mark(valueTparam, "T"),hoverdef("T", valueTparam)
	Q   int //@mark(valueQfield, "Q"),hoverdef("Q", valueQfield)
}
```
-- valueQfield-hoverdef --
```go
field Q int
```

\@mark\(valueQfield, \"Q\"\),hoverdef\(\"Q\", valueQfield\)
-- ValueTparam-hoverdef --
```go
type Value[T any] struct {
	val T   //@mark(ValueTparam, "T"),hoverdef("T", ValueTparam)
	Q   int //@mark(ValueQfield, "Q"),hoverdef("Q", ValueQfield)
}
```
-- ValueQfield-hoverdef --
```go
field Q int
```

[`(hover.Value).Q` on pkg.go.dev](https://pkg.go.dev/golang.org/x/tools/internal/lsp/godef/hover_generics?utm_source=gopls#Value.Q)

\@mark\(ValueQfield, \"Q\"\),hoverdef\(\"Q\", ValueQfield\)
